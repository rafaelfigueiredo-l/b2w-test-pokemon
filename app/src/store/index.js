import { createStore } from "redux";

const INITIAL_STATE = {
  theme: {},
  pokemon: [],
};
function course(state = INITIAL_STATE, action) {
  switch (action.type) {
    case "SET_THEME":
      return {
        ...state,
        theme: action.payload,
      };

    case "SET_POKEMON":
      return {
        ...state,
        pokemon: action.payload,
      };

    default:
      return state;
  }
}

const store = createStore(
  course,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
