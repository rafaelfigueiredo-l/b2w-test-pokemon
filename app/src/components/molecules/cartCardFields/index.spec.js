import { Provider } from "react-redux";
import store from "../../../store";
import CartFields from "./";

describe("<CartFields />", () => {
  const component = (
    <Provider store={store}>
      <CartFields />
    </Provider>
  );

  const mounted = mount(component);

  it("should mount correctly CartFields component", () => {
    expect(mounted.find(".cartFields")).toHaveLength(2);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly CartFields component", () => {
    expect(mounted.unmount()).toEqual({});
  });
});
