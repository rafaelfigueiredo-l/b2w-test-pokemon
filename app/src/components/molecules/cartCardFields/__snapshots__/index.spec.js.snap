// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartFields /> should mount correctly CartFields component 1`] = `
<styled.div
  className="cartFields"
>
  <div
    className="sc-AxhCb ccMvAG cartFields"
  >
    <p>
      <styled.label
        className="cartFieldLabel"
      >
        <label
          className="sc-AxjAm kizGHW cartFieldLabel"
        >
          Adicione o número do seu Cartão:
        </label>
      </styled.label>
      <styled.input
        className="cardFieldInput"
        onChange={[Function]}
        type="text"
        value=""
      >
        <input
          className="sc-AxirZ eSWeaK cardFieldInput"
          onChange={[Function]}
          type="text"
          value=""
        />
      </styled.input>
    </p>
    <p>
      <styled.input
        checked={0}
        className="cardFieldCheckBox"
        onChange={[Function]}
        placeholder="Lembrar cartão"
        type="checkbox"
      >
        <input
          checked={0}
          className="sc-AxiKw krEJia cardFieldCheckBox"
          onChange={[Function]}
          placeholder="Lembrar cartão"
          type="checkbox"
        />
      </styled.input>
      <styled.label
        className="cartFieldLabel"
      >
        <label
          className="sc-AxjAm kizGHW cartFieldLabel"
        >
          Lembrar Cartão
        </label>
      </styled.label>
    </p>
  </div>
</styled.div>
`;
